appscale/tools/agents/azure_agent.py

```diff
@@ -829,16 +978,40 @@ def delete_vmss_instance(self, compute_client, parameters, vmss_name, instance_i
     """
     resource_group = parameters[self.PARAM_RESOURCE_GROUP]
     verbose = parameters[self.PARAM_VERBOSE]
-    AppScaleLogger.verbose("Deleting Virtual Machine Instance {0} from Scale "
-      "Set {1} ...".format(instance_id, vmss_name), verbose)
+
+    vm_info = "Virtual Machine {0} from Scale Set {1}".format(instance_id,
+                                                              vmss_name)
+
+    # Check if we succeeded deleting the instance before but timed out.
+    vm_list = compute_client.virtual_machine_scale_set_vms.list(
+        resource_group, vmss_name)
+    already_deleted = True
+    for vm in vm_list:
+      if instance_id == vm.instance_id:
+        already_deleted = False
+        break
+    if already_deleted:
+      AppScaleLogger.verbose("{0} has already been deleted".format(vm_info),
+                             verbose)
+      return
+
+    AppScaleLogger.verbose("Deleting {0} ...".format(vm_info), verbose)
     result = compute_client.virtual_machine_scale_set_vms.delete(resource_group,
                                                                  vmss_name,
                                                                  instance_id)
     resource_name = 'Virtual Machine Instance ' + instance_id
     self.sleep_until_delete_operation_done(result, resource_name,
                                            self.MAX_VM_UPDATE_TIME, verbose)
-    AppScaleLogger.verbose("Virtual Machine Instance {0} from Scale Set {1} "
-      "has been successfully deleted".format(instance_id, vmss_name), verbose)
+    # Double check if we succeeded deleting the instance.
+    vm_list = compute_client.virtual_machine_scale_set_vms.list(
```

Comment:
Is this really necessary? Is it a known problem for the `delete` call to succeed and the instance to survive?

Reply:
I have seen it print out that we successfully deleted an instance when the instance is still there.